lib/src/package_analysis/common.dart

```diff
@@ -1,12 +1,235 @@
-import 'package:analyzer/dart/analysis/analysis_context.dart';
-import 'package:analyzer/dart/analysis/analysis_context_collection.dart';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:analyzer/dart/analysis/session.dart';
+import 'package:analyzer/file_system/file_system.dart' as resource;
+import 'package:pana/pana.dart';
+import 'package:path/path.dart' as path;
+import 'package:pub_semver/pub_semver.dart';
+import 'package:pubspec_parse/pubspec_parse.dart' hide Pubspec;
+
+const indentedEncoder = JsonEncoder.withIndent('  ');
 
 abstract class PackageAnalysisContext {
-  AnalysisContextCollection get analysisContextCollection;
+  AnalysisSession get analysisSession;
+
+  /// The name of the package being analyzed.
+  late final String packageName = pubspec.name;
+
+  /// The name of the singular dependency of this target dummy package, or null
+  /// if the package being analysed does not use a dummy.
+  final String? targetPackageName;
```

Comment:
How is this different from `packageName`, do we ever need to know `packageName` after the initial setup (if these two are different?)

Reply:
addressed in https://github.com/CicadaCinema/pana/pull/2/commits/7e09995c47283da2c271325567dc918abe7f00c8